tests/autoyast/installation.pm

```diff
@@ -127,8 +128,17 @@ sub verify_timeout_and_check_screen {
 sub run {
     my ($self) = @_;
 
+    my $test_data = get_test_suite_data();
+
     my @needles = qw(bios-boot nonexisting-package reboot-after-installation linuxrc-install-fail scc-invalid-url warning-pop-up autoyast-boot package-notification nvidia-validation-failed);
     my $expected_licenses = get_var('AUTOYAST_LICENSE');
+    my (@expected_warnings, @actual_warnings);
+    if (defined $test_data->{expected_warnings}) {
```

Comment:
if another AutoYaST test suite does not have `test_data`, it would crash, right? `undef->{expected_warning}`

Reply:
ahm, defined check from left to right or from right to left? I mean, to access a hash key from a hash that is undefined I would expect failure there, but maybe not, did you try with another test suite or know this rule?

Reply:
It works: e.g. http://10.160.66.6/tests/1337

Reply:
interesting Perl thing :) thanks for testing it. I will also be using that

Reply:
I guess it will even work with `if ($test_data->{expected_warnings}) {` due to [Perl vivification](https://perlmaven.com/autovivification).